src/main/java/ir/sahab/nexus/plugin/tag/internal/TagRestResource.java

```diff
@@ -0,0 +1,150 @@
+package ir.sahab.nexus.plugin.tag.internal;
+
+import static org.sonatype.nexus.rest.APIConstants.V1_API_PREFIX;
+
+import ir.sahab.nexus.plugin.tag.internal.dto.Tag;
+import ir.sahab.nexus.plugin.tag.internal.dto.TagCloneRequest;
+import ir.sahab.nexus.plugin.tag.internal.dto.TagDefinition;
+import ir.sahab.nexus.plugin.tag.internal.exception.ErrorResponse;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validator;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.sonatype.goodies.common.ComponentSupport;
+import org.sonatype.nexus.rest.Resource;
+
+/**
+ * Endpoint which provides RESTful API for tagging.
+ */
+@Named
+@Singleton
+@Path(V1_API_PREFIX + "/tags")
+public class TagRestResource extends ComponentSupport implements Resource, TagRestResourceDoc {
+
+    private final TagStore tagStore;
+    private final Validator validator;
+
+    @Inject
+    public TagRestResource(TagStore tagStore, Validator validator) {
+        this.tagStore = tagStore;
+        this.validator = validator;
+    }
+
+    @GET
+    @Path("/{name}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Override
+    public Tag getByName(@PathParam("name") String name) {
+        return tagStore.getByName(name);
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Override
+    public List<Tag> list(@QueryParam("attributes") List<String> attributes) {
```

Comment:
I have these reasons for this desing:
1.Having URL /tags/attributes is not compatible with resourceful view of endpoints. I tried to be resourceful instead of having different endpoints for different actions
2. I will add another attributes called components which will be components to search on requests and this way we will have combination of these filters as well.
3. Having static query parameters instead of dynamic has the advantage that swagger will create good form for it, in this way u can easily add multiple attribute search in UI by using + button.